src/rt-app_parse_config.c

```diff
@@ -526,27 +561,27 @@ parse_task_event_data(char *name, struct json_object *obj,
 	if (!strncmp(name, "timer", strlen("timer"))) {
 
 		tmp = get_string_value_from(obj, "ref", TRUE, "unknown");
-		if (!strncmp(tmp, "unique", strlen("unique")))
-				ref = create_unique_name(unique_name, sizeof(unique_name), tmp, tag);
-		else
-				ref = tmp;
 
-		i = get_resource_index(ref, rtapp_timer, opts);
+		if (!strncmp(tmp, "unique", strlen("unique"))) {
+			ref = create_unique_name(unique_name, sizeof(unique_name), tmp, tag);
+			resources_table = &tdata->local_resources;
+			data->type = rtapp_timer_unique;
```

Comment:
Do we need a different type? Can't we make timers local even for tasks that don't fork?

Reply:
not sure to get your point. "unique" is already a key of rt-app timer description
Any timer with a "unique" name will be unique/local to each thread whatever the thread is created at the beg or forked later.

Reply:
You actually answered my doubt in question above.. there are cases where tasks need to share timers.. so we need that distinction.